tensorflow/java/BUILD

```diff
@@ -311,9 +311,11 @@ tf_cc_test(
     srcs = [
         "src/gen/cc/source_writer_test.cc",
     ],
+    data = [
+        "src/gen/resources/test.snippet.java",
```

Comment:
Can we inline this one line string in the test instead of adding a data dependency?

Reply:
The only reason for having snippet is to allow Java source code to be read from a file instead of inlining it in a C++ string. It is not required, but could be convenient in some cases (in the next PR, you'll see that the Google licence header added to Java source files is in a snippet, so it can be easily found and updated if required).

That being said, for testing writing a snippet, we do need to keep it in a file, otherwise the test is obsolete. I think I'll add a new shortcut method in SourceWriter to write a snippet directly, something like `source_writer.write(snippet)`, it looks clean.

Reply:
Finally, I have removed the whole `Snippet` type in `java_defs.h` and added a utility method `WriteFromFile()` in `SourceWriter` instead.